server/service/IndividualServicesService.js

```diff
@@ -59,43 +62,58 @@ exports.bequeathYourDataAndDie = function (body, user, originator, xCorrelator,
        ****************************************************************************************/
       let applicationName = body["new-application-name"];
       let releaseNumber = body["new-application-release"];
+      let applicationProtocol = body["new-application-protocol"];
       let applicationAddress = body["new-application-address"];
       let applicationPort = body["new-application-port"];
 
       /****************************************************************************************
        * Prepare logicalTerminatinPointConfigurationInput object to 
        * configure logical-termination-point
        ****************************************************************************************/
+      
+      const HttpClientLtpUuidFromForwarding = await resolveHttpClient('PromptForBequeathingDataCausesNewTARbeingRequestedToRedirectInfoAboutApprovals');
+      if (HttpClientLtpUuidFromForwarding == undefined) {
+        reject(new Error(`The NewRelease ${applicationName} was not found.`));
+        return;
+      }
+
       let isdataTransferRequired = true;
-      let newReleaseUuid = await httpClientInterface.getHttpClientUuidAsync("NewRelease");
-      let currentApplicationName = await httpServerInterface.getApplicationNameAsync();
-      if (currentApplicationName == applicationName) {
-        let isUpdated = await httpClientInterface.setReleaseNumberAsync(newReleaseUuid, releaseNumber);
-        let currentApplicationRemoteAddress = await TcpServerInterface.getLocalAddress();
-        let currentApplicationRemotePort = await TcpServerInterface.getLocalPort();
-        if((applicationAddress == currentApplicationRemoteAddress) && 
-        (applicationPort == currentApplicationRemotePort)){
-          isdataTransferRequired = false;
+      let logicalTerminationPointConfigurationStatus = {};
+      let newReleaseHttpClientLtpUuid = HttpClientLtpUuidFromForwarding[0];
+      if (newReleaseHttpClientLtpUuid != undefined) {
+      
+        let isReleaseUpdated = await httpClientInterface.setReleaseNumberAsync(newReleaseHttpClientLtpUuid, releaseNumber);
```

Comment:
As set operation will always return true if write to the database is successful and it will not check for the value update . We have to include a check before set operation if incoming value is different from the value present if so then set the value. In this way we can assure wrong forwarding is not sent to ALT. Please include this change for all set attribute methods.

Reply:
done